releasetool/commands/start/go.py

```diff
@@ -0,0 +1,206 @@
+
+
+def determine_release_version(ctx: Context) -> None:
+    click.secho(f"> Now it's time to pick a release version!", fg="cyan")
+    release_notes = textwrap.indent(ctx.release_notes, "\t")
+    click.secho(f"Here's the release notes you wrote:\n\n{release_notes}\n")
+
+    parsed_version = [int(x) for x in ctx.last_release_version.split(".")]
+
+    if parsed_version == [0, 0, 0]:
+        ctx.release_version = "0.1.0"
+        return
+
+    selection = click.prompt(
+        "Is this a major, minor, or patch update (or enter the new version " "directly)"
+    )
+    if selection == "major":
+        parsed_version[0] += 1
+        parsed_version[1] = 0
+        parsed_version[2] = 0
+    elif selection == "minor":
+        parsed_version[1] += 1
+        parsed_version[2] = 0
+    elif selection == "patch":
+        parsed_version[2] += 1
+    else:
+        ctx.release_version = selection
+        return
+
+    ctx.release_version = "{}.{}.{}".format(*parsed_version)
+
+    click.secho(f"Got it, releasing {ctx.release_version}.")
+
+
+def create_release_branch(ctx) -> None:
+    if ctx.module_name is None:
+        ctx.release_branch = f"release-{ctx.release_version}"
+    else:
+        ctx.release_branch = f"release-{ctx.module_name}-{ctx.release_version}"
+    click.secho(f"> Creating branch {ctx.release_branch}", fg="cyan")
+    return releasetool.git.checkout_create_branch(ctx.release_branch)
+
+
+def update_changelog(ctx: Context) -> None:
+    click.secho(f"> Updating {_CHANGELOG_FILENAME}.", fg="cyan")
+
+    if not os.path.exists(_CHANGELOG_FILENAME):
+        print(f"{_CHANGELOG_FILENAME} does not yet exist. Opening it for " "creation.")
+
+        releasetool.filehelpers.open_editor_with_content(
+            _CHANGELOG_FILENAME, _CHANGELOG_TEMPLATE
+        )
+
+    changelog_entry = f"## v{ctx.release_version}" f"\n\n" f"{ctx.release_notes}" f"\n\n"
+    releasetool.filehelpers.insert_before(
+        _CHANGELOG_FILENAME, changelog_entry, "^## (.+)$|\Z"
+    )
+
+def create_release_commit(ctx: Context) -> None:
+    """Create a release commit."""
+    click.secho("> Comitting changes", fg="cyan")
+    releasetool.git.commit(
+        [_CHANGELOG_FILENAME], f"Release {ctx.release_version}"
+    )
+
+def create_release_cl(ctx: Context) -> None:
+    click.secho(f"> Creating release CL.", fg="cyan")
+    revs = click.prompt("reviewers (comma-separated)", default="deklerk,jba")
+    subprocess.check_output(["git", "codereview", "mail", "-r", revs, "HEAD"])
+
+
+def start() -> None:
+    ctx = Context()
+
+    click.secho(f"o/ Hey, {getpass.getuser()}, let's release some stuff!", fg="magenta")
+
+    determine_module_name(ctx)
+    determine_last_release(ctx)
+    gather_changes(ctx)
+    releasetool.commands.common.edit_release_notes(ctx)
+    determine_release_version(ctx)
+    create_release_branch(ctx)
+    update_changelog(ctx)
+    create_release_commit(ctx)
+    create_release_cl(ctx)
```

Comment:
Should there be one more step that instructs the user to push a tag?

Reply:
That is a separate command, `releasetool tag`. It can't be in the same command because someone has to approve the CL/PR.

Reply:
Ah ok 👍